src/tribol/mesh/MfemData.cpp

```diff
@@ -891,6 +915,221 @@ std::unique_ptr<mfem::BlockOperator> MfemJacobianData::GetMfemBlockJacobian( con
+                               static_cast<int>( BlockSpace::LAGRANGE_MULTIPLIER ) ) );
+  submesh_J.Finalize();
+  submesh_J_hypre = GetUpdateData().submesh_redecomp_xfer_01_->ConvertToHypreParMatrix( submesh_J, false );
+  // Matrix returned by mfem::ParMult copies row and column starts since last arg is true. All other data is copied and
+  // owned by the new matrix.
+  parent_J_hypre = std::unique_ptr<mfem::HypreParMatrix>(
+      mfem::ParMult( std::unique_ptr<mfem::HypreParMatrix>( submesh_parent_vdof_xfer_->Transpose() ).get(),
+                     submesh_J_hypre.get(), true ) );
+  block_J->SetBlock( 0, 1,
+                     mfem::RAP( parent_data_.GetParentCoords().ParFESpace()->Dof_TrueDof_Matrix(), parent_J_hypre.get(),
+                                submesh_data_.GetSubmeshFESpace().Dof_TrueDof_Matrix() ) );
+
+  // transfer (1, 0) block (gap dof rows, displacement dof cols)
+  submesh_J = GetUpdateData().submesh_redecomp_xfer_10_->TransferToParallelSparse(
+      lm_elems, mortar_elems,
+      method_data.getBlockJ()( static_cast<int>( BlockSpace::LAGRANGE_MULTIPLIER ),
+                               static_cast<int>( BlockSpace::MORTAR ) ) );
+  submesh_J += GetUpdateData().submesh_redecomp_xfer_10_->TransferToParallelSparse(
+      lm_elems, nonmortar_elems,
+      method_data.getBlockJ()( static_cast<int>( BlockSpace::LAGRANGE_MULTIPLIER ),
+                               static_cast<int>( BlockSpace::NONMORTAR ) ) );
+  submesh_J.Finalize();
+  submesh_J_hypre = GetUpdateData().submesh_redecomp_xfer_10_->ConvertToHypreParMatrix( submesh_J, false );
+  // Matrix returned by mfem::ParMult copies row and column starts since last arg is true. All other data is copied and
+  // owned by the new matrix.
+  parent_J_hypre = std::unique_ptr<mfem::HypreParMatrix>( std::unique_ptr<mfem::HypreParMatrix>(
+      mfem::ParMult( submesh_J_hypre.get(), submesh_parent_vdof_xfer_.get(), true ) ) );
+  block_J->SetBlock( 1, 0,
+                     mfem::RAP( submesh_data_.GetSubmeshFESpace().Dof_TrueDof_Matrix(), parent_J_hypre.get(),
+                                parent_data_.GetParentCoords().ParFESpace()->Dof_TrueDof_Matrix() ) );
+
+  // Create ones on diagonal of eliminated mortar tdofs (CSR sparse matrix -> HypreParMatrix) I vector
+  auto& submesh_fes = submesh_data_.GetSubmeshFESpace();
+  mfem::Array<int> rows( submesh_fes.GetTrueVSize() + 1 );
+  rows = 0;
+  auto mortar_tdofs_ct = 0;
+  for ( int i{ 0 }; i < submesh_fes.GetTrueVSize(); ++i ) {
+    if ( mortar_tdofs_ct < mortar_tdof_list_.Size() && mortar_tdof_list_[mortar_tdofs_ct] == i ) {
+      ++mortar_tdofs_ct;
+    }
+    rows[i + 1] = mortar_tdofs_ct;
+  }
+  // J vector
+  mfem::Array<int> mortar_tdofs( mortar_tdof_list_ );
+  // data vector
+  mfem::Vector ones( mortar_tdofs_ct );
+  ones = 1.0;
+  mfem::SparseMatrix inactive_sm( rows.GetData(), mortar_tdofs.GetData(), ones.GetData(), submesh_fes.GetTrueVSize(),
```

Comment:
Add comment to clarify what "inactive" means here in terms of jacobian contributions.

Reply:
added